.github/workflows/call_test_package.yml

```diff
@@ -0,0 +1,142 @@
+         --md-report --md-report-output=test-results-${{env.REPORT_NAME}}.md\
+         --self-contained-html --html=report-${{env.REPORT_NAME}}.html\
+         --pytest-durations=10 --pytest-resultlog=log-${{env.REPORT_NAME}}.txt
+    ## apploading results
+    - name: Upload HTML test result
+      if: ${{ always() && !cancelled()}}
+      id: html_report
+      uses: actions/upload-artifact@v4
+      with:
+        name: pytest-results-html-${{env.REPORT_NAME}}
+        path: report-${{env.REPORT_NAME}}.html
+        if-no-files-found: error
+    # add the link for html report to the mardown file
+    - name: Save link HTML in file
+      if: ${{ always() && !cancelled()}}
+      run: 
+        echo -e "\n [HTML summary]("${{ steps.html_report.outputs.artifact-url }}")" >> test-results-${{env.REPORT_NAME}}.md
+    - name: Upload md test result 
+      if: ${{ always() && !cancelled()}}
+      uses: actions/upload-artifact@v4
+      with:
+        name: pytest-results-md-${{env.REPORT_NAME}}
+        path: test-results-${{env.REPORT_NAME}}.md 
+        if-no-files-found: error
+    - name: Upload duration test result
+      if: ${{ always() && !cancelled()}}
+      uses: actions/upload-artifact@v4
+      with:
+        name: pytest-results-log-${{env.REPORT_NAME}}
+        path: log-${{env.REPORT_NAME}}.txt
+        if-no-files-found: error
+    - name: Upload coverage data
+      if: ${{ always() && !cancelled()}}
+      uses: actions/upload-artifact@v4
+      with:
+        name: cov-results-${{env.REPORT_NAME}}
+        path: cov/cov-${{env.REPORT_NAME}}.xml
+        if-no-files-found: error
+
+  merge_md_test_data:
+    # for more details
+    # https://github.com/thombashi/pytest-md-report
+    needs: build-test
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    steps:
+    # get report from previous job 
+    - name: Get data
+      uses: actions/download-artifact@v4
+      with:
+        pattern: pytest-results-md-*-os-*
+    # create a comment in the PR with the failling tests
+    # for more details: https://github.com/marocchino/sticky-pull-request-comment
+    - name: Render the report to the PR when tests fail
+      uses: marocchino/sticky-pull-request-comment@v2
+      with:
+        header: test-report
+        recreate: true
+        path: ./pytest-results-md-3.13-os-macos-latest/test-results-3.13-os-macos-latest.md     #CHECK: need to be updated when new version are available
+    # create a comment in the summary with the failling tests
+    - name: Output reports to the job summary when tests fail
+      shell: bash
+      run: |
+        for folder in *;
+        do 
+          echo "<details><summary>Failed Test Report: $folder</summary>" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          cat $folder/*.md >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "</details>" >> $GITHUB_STEP_SUMMARY
+        done
+
+  merge_upload_coverage_data:
```

Comment:
no sure this is necessary: I think codecov does that for you

or least I don't ever remember having to 'manually' merge xml before sending them to codecov